docs: clarify native histograms visualization and mark as experimental (#9776)
grafana/mimir: docs/sources/mimir/visualize/native-histograms/_index.md

# Visualize native histograms

Native histograms are experimental and subject to change.

> Prometheus native histograms is a new data type beside the existing floating point time series in the Prometheus ecosystem that makes it possible to produce, store, and query a high-resolution [histogram](https://prometheus.io/docs/concepts/metric_types/#histogram) of observations.

Reviewer: Also, we should remove the `new` to keep the docs current over time.

> Prometheus native histograms is a new data type beside the existing floating point time series in the Prometheus ecosystem that makes it possible to produce, store, and query a high-resolution [histogram](https://prometheus.io/docs/concepts/metric_types/#histogram) of observations.

Reviewer: Okay, thank you for clarifying this. In this case, can we just remove the `beside the existing floating point time series` clause?

To learn more about the native histograms data type and how to start sending native histograms to Grafana Mimir,
refer to [Send native histograms to Mimir]({{< relref "../../send/native-histograms" >}}).

In general not all visualizations support displaying the new native histogram data type, however you can use the Prometheus Query Language (PromQL) to derive the usual floating point time series from native histograms that can be used in visualizations as before.

## Prometheus Query Language

The Prometheus Query Language (PromQL) allows you to query native histogram metrics. The data type of the result depends on the query and the underlying data.

For more information about PromQL, refer to [Querying Prometheus](https://prometheus.io/docs/prometheus/latest/querying/basics/).

> The following sections describe common ways to derive floating point time series from native histograms data for visualizations and also how to convert existing queries using classic histograms into queries using native histograms.

Reviewer: ```suggestion
The following examples show common ways to derive floating point time series from native histograms data for visualizations and also how to convert existing queries using classic histograms into queries using native histograms.
```

Reviewer: We try to avoid "documenting the documentation." Instead of `sections`, could we say `examples`, `queries`, or something more specific?

## Create Grafana dashboards

> The panel types [Histogram](/docs/grafana/latest/panels-visualizations/visualizations/histogram/) and [Heatmap](/docs/grafana/latest/panels-visualizations/visualizations/heatmap/) are compatible with the new native histogram data type. Use these to visualize a query such as:

Reviewer: ```suggestion
The panel types [Histogram](/docs/grafana/latest/panels-visualizations/visualizations/histogram/) and [Heatmap](/docs/grafana/latest/panels-visualizations/visualizations/heatmap/) are compatible with the native histogram data type. Use these panel types to visualize a query, such as:
```

```PromQL
sum(rate(request_duration_seconds[$__rate_interval]))
```

> For all other panel types, for example [Timeseries](/docs/grafana/latest/panels-visualizations/visualizations/time-series/), use one of the histogram functions above in [Prometheus Query Language](#prometheus-query-language) to visualize a derived float time series.

Reviewer: ```suggestion
For all other panel types, for example [Timeseries](/docs/grafana/latest/panels-visualizations/visualizations/time-series/), use one of the histogram functions in [Prometheus Query Language](#prometheus-query-language) to visualize a derived float time series.
```

## Grafana classic explore

> In [Explore](https://grafana.com/docs/grafana/latest/explore/), use one of the histogram functions above in [Prometheus Query Language](#prometheus-query-language) to visualize a derived float time series.

Reviewer: ```suggestion
In [Explore](https://grafana.com/docs/grafana/latest/explore/), use one of the histogram functions in [Prometheus Query Language](#prometheus-query-language) to visualize a derived float time series.
```

{{% admonition type="note" %}}

> Visualizing native histogram data type directly without the histogram functions in the **Explore** view is a work in progress.

Reviewer: Is it `native histogram` or `native histograms`? We need to be consistent throughout the doc.

Reviewer: Rather than saying it's a work in progress, we should say it's not supported at this time.
